tests/test__utils.py

```diff
@@ -0,0 +1,70 @@
+import functools
+
+from starlette._utils import iscoroutinefunction
+
+
+def test_async_func():
+    async def async_func():
+        ...  # pragma: no cover
+
+    def func():
+        ...  # pragma: no cover
+
+    assert iscoroutinefunction(async_func)
+    assert not iscoroutinefunction(func)
+
+
+def test_async_partial():
```

Comment:
Would it also make sense to test a nested partial to check the `while` loop?

```python
def test_async_nested_partial():
    async def async_func(a, b):
        ...

    def func(a, b):
        return partial(async_func, a=a)

    nested_partial = functools.partial(func, 1)
    assert iscoroutinefunction(nested_partial)
```

Reply:
Good idea! Thanks! :)

Reply:
The proposed snippet is not supported, because `func` is sync.

I'll add a test for nested one, tho.

Reply:
Ah, got it, my bad with the snippet, but you got the gist of it and corrected ;)